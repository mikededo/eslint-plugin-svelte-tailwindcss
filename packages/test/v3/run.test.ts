import Utils from '@eslint-plugin-svelte-tailwindcss/test-shared/utils';
import { spawnSync } from 'node:child_process';

describe('e2e', () => {
  it.each(['default', 'config', 'settings'] as const)('should lint with the %s configuration', async (type) => {
    const result = spawnSync(`TEST_TYPE=${type === 'default' ? '' : type} pnpm lint`, [], {
      encoding: 'utf-8',
      shell: true
    });

    expect(Utils.parseStdOut(result.stdout)).toMatchSnapshot('stdout');
    expect(result.status).toMatchSnapshot('exit code');
  });
});
