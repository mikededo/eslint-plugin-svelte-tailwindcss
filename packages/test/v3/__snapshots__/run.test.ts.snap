// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`e2e > should lint with the config configuration > exit code 1`] = `1`;

exports[`e2e > should lint with the config configuration > stdout 1`] = `
"
/eslint-plugin-svelte-tailwindcss/packages/test/v3
> eslint . -c eslint.config.test.mjs


/eslint-plugin-svelte-tailwindcss/packages/test/v3/src/routes/+page.svelte
   4:20  error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  17:6   error    TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  18:6   error    Do not mix literal expressions with mustache expressions  svelte-tailwindcss/no-literal-mustache-mix
  20:1   warning  Using '@apply' requires setting style lang to postcss     svelte-tailwindcss/at-apply-require-postcss

/eslint-plugin-svelte-tailwindcss/packages/test/v3/src/routes/styles.ts
   3:7  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes
  10:1  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

✖ 9 problems (8 errors, 1 warning)
  8 errors and 1 warning potentially fixable with the \`--fix\` option.

 ELIFECYCLE  Command failed with exit code 1.
"
`;

exports[`e2e > should lint with the default configuration > exit code 1`] = `1`;

exports[`e2e > should lint with the default configuration > stdout 1`] = `
"
/eslint-plugin-svelte-tailwindcss/packages/test/v3
> eslint . -c eslint.config.test.mjs


/eslint-plugin-svelte-tailwindcss/packages/test/v3/src/routes/+page.svelte
  17:6  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  18:6  error  Do not mix literal expressions with mustache expressions  svelte-tailwindcss/no-literal-mustache-mix
  20:1  error  Using '@apply' requires setting style lang to postcss     svelte-tailwindcss/at-apply-require-postcss

✖ 3 problems (3 errors, 0 warnings)
  3 errors and 0 warnings potentially fixable with the \`--fix\` option.

 ELIFECYCLE  Command failed with exit code 1.
"
`;

exports[`e2e > should lint with the settings configuration > exit code 1`] = `1`;

exports[`e2e > should lint with the settings configuration > stdout 1`] = `
"
/eslint-plugin-svelte-tailwindcss/packages/test/v3
> eslint . -c eslint.config.test.mjs


/eslint-plugin-svelte-tailwindcss/packages/test/v3/src/routes/+page.svelte
   4:20  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
   5:21  error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  17:6   error  TailwindCSS classes should be sorted                      svelte-tailwindcss/sort-classes
  18:6   error  Do not mix literal expressions with mustache expressions  svelte-tailwindcss/no-literal-mustache-mix
  20:1   error  Using '@apply' requires setting style lang to postcss     svelte-tailwindcss/at-apply-require-postcss

/eslint-plugin-svelte-tailwindcss/packages/test/v3/src/routes/styles.ts
   3:7  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes
  10:1  error  TailwindCSS classes should be sorted  svelte-tailwindcss/sort-classes

✖ 9 problems (9 errors, 0 warnings)
  9 errors and 0 warnings potentially fixable with the \`--fix\` option.

 ELIFECYCLE  Command failed with exit code 1.
"
`;
